site_scons/site_tools/antlr4.py

```diff
@@ -0,0 +1,102 @@
+import os
+import SCons.Util
+from SCons.Script import Builder, Action, Dir, File, Entry
+
+
+def _detect(env):
+    """Try to find the Antlr parser generator"""
+    try:
+        return env['ANTLR4']
+    except KeyError:
+        raise SCons.Errors.StopError(
+            'Could not find antlr4 parser generator')
+
+
+def _antlr4_emitter(target, source, env):
+    """Process sources and flags"""
+    target = []
+    antlr4_suffix = env.subst('$ANTLR4_SUFFIX')
+    antlr4_h_suffix = env.subst('$ANTLR4_HSUFFIX')
+    antlr4_cc_suffix = env.subst('$ANTLR4_CCSUFFIX')
+    antlr4_token_suffix = env.subst('$ANTLR4_TOKENSUFFIX')
+    if env['ANTLR4_FLAGS']:
+        antlr4flags = env.subst('$ANTLR4_FLAGS', target=target, source=source)
+        flags = SCons.Util.CLVar(antlr4flags)
+    else:
+        flags = SCons.Util.CLVar('')
+
+    # -o flag
+    if env['ANTLR4_OUT']:
+        env['ANTLR4_OUT'] = Dir(env['ANTLR4_OUT'])
+        flags.append('-o ${ANTLR4_OUT}')
+
+    # -lib flag
+    if env['ANTLR4_GLIB']:
+        env['ANTLR4_GLIB'] = Dir(env['ANTLR4_GLIB'])
+        flags.append('-lib ${ANTLR4_GLIB}')
+
+    # update antlr4 flags
+    env['ANTLR4_FLAGS'] = str(flags)
```

Comment:
Hmm, I think this is very bad idea, but it is copied from old Antlr tool so it is probably acceptable for now (until Fritz writes correct CMake version of it 😸).